src/parser/ToInteger.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_PARSER_TOINTEGER_H
+#define SQUID_PARSER_TOINTEGER_H
+
+#include "sbuf/SBuf.h"
+#include "sbuf/Stream.h"
+
+#include <limits>
+
+namespace Parser
+{
+
+namespace Impl
```

Comment:
Unfortunately it looks like we [cannot](https://stackoverflow.com/questions/357564/are-there-any-uses-for-unnamed-namespaces-in-header-files ) use an unnamed namespace here with their static linkage limitations. Also with this 'internal' namespace we can probably drop the '_' suffix in its function names.